src/js/components/Pagination/__tests__/Pagination-test.tsx

```diff
@@ -444,6 +444,46 @@ describe('Pagination', () => {
     expect(updatedSelectButton).toBeTruthy();
   });
 
+  test('should respect stepOptions and controlled step', async () => {
+    window.scrollTo = jest.fn();
+    const user = userEvent.setup();
```

Comment:
Related to our discussion yesterday, because we are using both fireEvent and userEvent in the same test script then we will receive test timeouts when running in parallel (e.g. `yarn test-watch`). This does not occur when using `--runInBand` which executes the tests serially.

Let's chat whether we want to refactor all tests in Pagination to use either fireEvent or userEvent. Or, explore the leads below.

Potentially a couple leads here:
https://github.com/testing-library/user-event/issues/833

https://github.com/testing-library/user-event/issues/833#issuecomment-1171452841

```
Using userEvent instead of fireEvent in the same test script can cause performance issues and timeout errors in the Testing Library. This is because userEvent provides more advanced browser interaction simulation than fireEvent, which can overcome the default Jest timeout limit of 5 seconds per test. 

github.com
Performance issues · testing-library user-event · Discussion #1144
Recently, in one of new tests, I've noticed that it often overcomes the default Jest timeout limit of 5s per test. After investigating for a while, I've discovered that the performance decrease was unfortunately caused by using userEvent instead of fireEvent. An example: a simple test ...

testing-library.com
user-event v13 - Testing Library
Nov 23, 2023 — user-event is a companion library for Testing Library that provides more advanced simulation of browser interactions than the built-in fireEvent method.
One solution to this issue is to set the delay option on the userEvent to null. For example, const user = userEvent.setup({ delay: null });. 
Here are some other tips for using the Testing Library:
Avoid wrapping fireEvent in act or await waitFor.
Use the async RTL methods that return promises and are resolved when an element is found. 
The Testing Library is a lightweight solution for testing that provides utilities for querying for nodes. It can fire events like a button click or a form submit using the fireEvent function.
```

Reply:
I tried refactoring the code to all use `userEvent`, but now I'm getting errors where onChange is not being called the correct number of times. Seems as though counts are bleeding over from other tests.

Should we use fireEvent for now or just leave as is with mix of fireEvent/userEvent and improve later?

Reply:
Going to leave as is for now.